### ISSUE TITLE
Multiple projects in one window

### ISSUE THREAD

**Issue by probablykasper** [zed-industries/zed#4948]
### Check for existing issues

- [X] Completed

### Describe the feature

In VSCode I usually have multiple projects open at the same time, and it's convenient for me to have them all in one window. In VSCode I use the `New Window Tab` command to open native macOS tabs:
<img width="1109" alt="image" src="https://user-images.githubusercontent.com/11315492/225148476-334a9518-eeaa-43c8-9baf-89d9408c0d2e.png">

Would be great to have a solution for this in Zed (doesn't need to be native window tabs)

**Comment by andrewthauer** [comment]
I believe this is called native tabs in macOS. You can enable the desired behaviour for macOS (which vscode respects) in `Settings > Desktop & Dock > Windows > Prefer tabs when opening documents = Always`

**Comment by d-Pixie** [comment]
This is currently one of two deal breakers for me (the other one is language support, but that is fixable), so I would love to see it supported 🙂

@andrewthauer yes that is correct, but Zed currently doesn't respect this and still gives you separate windows.

**Comment by d-Pixie** [comment]
I read this, after checking out the VS code source: https://worthdoingbadly.com/vscodetabs/. If that is of any help to the devs. Unfortunately neither OS X coding or Rust is in my toolset, so I can't PR this one ...

**Comment by nzbasic** [comment]
I also love this feature of VSCode and have been hoping for its addition in Zed ever since I downloaded it. It has been a while so I wanted to give it a go myself.. forewarning I have never used Rust before and am very unfamiliar with the Zed codebase

After finding this stack overflow answer: https://stackoverflow.com/questions/48254397/nswindow-controlling-multiple-tabs/49832441#49832441

Led me to this: https://developer.apple.com/documentation/appkit/nswindow/1644729-tabbingmode?language=objc

The values for that enum seem to be:
```
Automatic = 0
Preferred = 1
Disallowed = 2
```

So I added the line
`let _: () = msg_send![native_window, setTabbingMode: 1];` 
at line 629 of:
`crates/gpui/src/platform/mac/window.rs`

Which resulted in this:
<img width="1024" alt="image" src="https://github.com/user-attachments/assets/3e3506ac-147f-41c7-bb1b-95f528a1e84e">

A bit ugly, but it works! My naive solution for this was to just add some arbitrary padding to the title bar at `crates/title_bar/src/title_bar.rs`. However when there is only one tab in a window it looks pretty bad.
<img width="364" alt="image" src="https://github.com/user-attachments/assets/6dc49126-9062-4037-9925-e87c5607409f">

Here is the final result with multiple tabs in one window, I am able to click in each to open them, I can open files, create a terminal etc and it all seems isolated:
<img width="2134" alt="image" src="https://github.com/user-attachments/assets/985e302a-2de5-4283-a37f-93f212f72203">

Seems like a pretty basic change that is well supported by the OS, it also unlocks these native options at the bottom in window:
<img width="295" alt="image" src="https://github.com/user-attachments/assets/d77fb2b4-61e1-435d-bccf-ee0cf693e33f">

I am also able to drag the tabs to join Zed windows together

Hopefully someone more familiar with Rust and Zed code can help to give support for this feature with more polish

For reference I also saw this related issue https://github.com/zed-industries/zed/issues/7136

**Comment by s123121** [comment]
This and better git support are the 2 blocker keeping me to switch my workflow entirely to zed

**Comment by danielfigueiredo** [comment]
Adding more folders to the existing workspace isn't ideal, because now all document search and everything is mixed up
You can't bounce between workspaces using a single unified window (like on neovim, emacs, etc) 
I work with multiple repositories every day, I was very excited with Zed but I can't be cmd + ` several times every time I need to find my other workspace. I would expect to hit a cmd, type the name and select from a dropdown or something

**Comment by fmunteanu** [comment]
This is a deal breaker for myself also, we multiple tabs into same editor window. Until then, I will continue to use Code.

**Comment by dalisoft** [comment]
I would love to see this feature

**Comment by timurguseynov** [comment]
@nzbasic is it working for you? Could you send a code/lines affected or a patch? Would love to use that, as it's a lot easier to navigate between projects.

**Comment by timurguseynov** [comment]
Zed is probably the best editor, but I still can't completely switch from Sublime.

The native tab feature allows me to quickly open multiple projects and see them all in a single bar. It makes switching between different projects incredibly easy—no need to use Alt + Tab or any other tool. Just glance at the bar and click, which is extremely useful when working on multiple tasks, opening different files, downloading and cloning projects, editing /etc/hosts, and basically using it as a tool for everything.

**Comment by yistc** [comment]
Would love to see this feature for working on multiple projects at the same time

**Comment by kaguya3222** [comment]
I would love to see this feature in Zed, one of the blockers to completely switch from Vscode

**Comment by danielfigueiredo** [comment]
One thing I have been doing to get around this issue is to use the recent projects navigation window, it pops up with a search bar and allows you to either open a new window or open in the current. This is a life saver, since I can now navigate around without using the mouse.
The only thing it cannot do is open a project you have never open before, in which case I go through command line `zed /path/to/new/project` 

Because of this I honestly do not miss this feature so much anymore, if Zed had the same thing for searching through open tabs and jumping to them it would be phenomenal!

**Comment by remorses** [comment]
There is already an implementation for this feature on Mac which would help many Zed users: https://github.com/zed-industries/zed/pull/20557

Is there any chance that PR can be re-opened?

**Comment by testpassword** [comment]
Waiting this feature so much.

**Comment by remorses** [comment]
There is a new PR open to land native Mac tabs support to Zed: https://github.com/zed-industries/zed/pull/32238
